pallets/loans/src/lib.rs

```diff
@@ -515,6 +543,30 @@ pub mod module {
             )?;
             Ok(().into())
         }
+
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn add_reserves(
+            origin: OriginFor<T>,
+            currency_id: CurrencyId,
+            add_amount: Balance,
+        ) -> DispatchResultWithPostInfo {
+            let root = ensure_signed(origin)?;
+            Self::add_reserves_internal(root, currency_id, add_amount)?;
+            Ok(().into())
+        }
+
+        #[pallet::weight(10_000)]
+        #[transactional]
+        pub fn reduce_reserves(
+            origin: OriginFor<T>,
+            currency_id: CurrencyId,
+            reduce_amount: Balance,
+        ) -> DispatchResultWithPostInfo {
+            let root = ensure_signed(origin)?;
```

Comment:
Use ensure_origin.

Reply:
I'd like to know how can I get the root account?

Reply:
Currently it is not verifying whether it's a root user. We can do one thing

```
ensure_root(origin)?;
let root = ensure_signed(origin)?;
```

We can verify and fetch root user.

Reply:
It will meet the error 
```
thread 'tests::reduce_reserves_works' panicked at 'Expected Ok(_). Got Err(
    DispatchErrorWithPostInfo {
        post_info: PostDispatchInfo {
            actual_weight: None,
            pays_fee: Pays::Yes,
        },
        error: BadOrigin,
    },
)', pallets/loans/src/tests.rs:385:9
```

Reply:
`Root` and `Signed(AccountId)` are different items in `RawOrigin`.

Reply:
oh yes, you are right @yz89